src/ros_filter.cpp

```diff
@@ -604,7 +604,7 @@ void RosFilter<T>::integrateMeasurements(const rclcpp::Time & current_time)
       const std::string first_measurement_topic =
         first_measurement->topic_name_;
       // revertTo may invalidate first_measurement
-      if (!revertTo(first_measurement_time - rclcpp::Duration(1))) {
+      if (!revertTo(first_measurement_time - rclcpp::Duration(1ns))) {
```

Comment:
@ayrton04 should this be 1ns, 1s, ...?

Reply:
Thanks, I meant to ask the same question.

Reply:
If I understand the equivalent ROS 1 code, it is also subtracting 1 nanosecond:

https://github.com/cra-ros-pkg/robot_localization/blob/a2da33bb3aaa4b10358de391221fe0ed97b16f49/src/ros_filter.cpp#L611

Reply:
So the PRs involved here are

https://github.com/cra-ros-pkg/robot_localization/pull/266/files

and

https://github.com/cra-ros-pkg/robot_localization/pull/279/files

The only reason we subtract any time at all is to ensure that we go to a measurement (presumably the first) that is before `first_measurement_time`. But we could also just change this to use `>=`:

https://github.com/cra-ros-pkg/robot_localization/blob/3326e80a4169365b4570b5adcd1a63f3e37ab54f/src/ros_filter.cpp#L3322-L3323

If I had to guess (I didn't dig too far), the measurement times were likely stored as doubles originally, and the `1e-9` thing was just me being careful with floating point numbers. But ROS times have exact representations, so I think the 1e-9 could go, and we can change the `revertTo` reverse iteration to check for `>=` instead of just `>`.

Reply:
However, that change could be outside of scope for this. It was meant to be 1 nanosecond, yes.